src/decrypter.js

```diff
@@ -21,99 +21,25 @@ const ntoh = function(word) {
- * @see http://en.wikipedia.org/wiki/Block_cipher_mode_of_operation#Cipher_Block_Chaining_.28CBC.29
- * @see https://tools.ietf.org/html/rfc2315
- */
-export const decrypt = function(encrypted, key, initVector) {
-  // word-level access to the encrypted bytes
-  let encrypted32 = new Int32Array(encrypted.buffer,
-                                   encrypted.byteOffset,
-                                   encrypted.byteLength >> 2);
-
-  let decipher = new AES(Array.prototype.slice.call(key));
-
-  // byte and word-level access for the decrypted output
-  let decrypted = new Uint8Array(encrypted.byteLength);
-  let decrypted32 = new Int32Array(decrypted.buffer);
-
-  // temporary variables for working with the IV, encrypted, and
-  // decrypted data
-  let init0;
-  let init1;
-  let init2;
-  let init3;
-  let encrypted0;
-  let encrypted1;
-  let encrypted2;
-  let encrypted3;
-
-  // iteration variable
-  let wordIx;
-
-  // pull out the words of the IV to ensure we don't modify the
-  // passed-in reference and easier access
-  init0 = initVector[0];
-  init1 = initVector[1];
-  init2 = initVector[2];
-  init3 = initVector[3];
-
-  // decrypt four word sequences, applying cipher-block chaining (CBC)
-  // to each decrypted block
-  for (wordIx = 0; wordIx < encrypted32.length; wordIx += 4) {
-    // convert big-endian (network order) words into little-endian
-    // (javascript order)
-    encrypted0 = ntoh(encrypted32[wordIx]);
-    encrypted1 = ntoh(encrypted32[wordIx + 1]);
-    encrypted2 = ntoh(encrypted32[wordIx + 2]);
-    encrypted3 = ntoh(encrypted32[wordIx + 3]);
-
-    // decrypt the block
-    decipher.decrypt(encrypted0,
-                     encrypted1,
-                     encrypted2,
-                     encrypted3,
-                     decrypted32,
-                     wordIx);
-
-    // XOR with the IV, and restore network byte-order to obtain the
-    // plaintext
-    decrypted32[wordIx] = ntoh(decrypted32[wordIx] ^ init0);
-    decrypted32[wordIx + 1] = ntoh(decrypted32[wordIx + 1] ^ init1);
-    decrypted32[wordIx + 2] = ntoh(decrypted32[wordIx + 2] ^ init2);
-    decrypted32[wordIx + 3] = ntoh(decrypted32[wordIx + 3] ^ init3);
-
-    // setup the IV for the next round
-    init0 = encrypted0;
-    init1 = encrypted1;
-    init2 = encrypted2;
-    init3 = encrypted3;
-  }
-
-  return decrypted;
-};
-
-/**
  * The `Decrypter` class that manages decryption of AES
  * data through `AsyncStream` objects and the `decrypt`
  * function
  *
  * @param {Uint8Array} encrypted the encrypted bytes
- * @param {Uint32Array} key the bytes of the decryption key
+ * @param {Uint8Array} key the bytes of the decryption key
  * @param {Uint32Array} initVector the initialization vector (IV) to
  * @param {Function} done the function to run when done
  * @class Decrypter
  */
 export class Decrypter {
   constructor(encrypted, key, initVector, done) {
+    let view = new DataView(key.buffer);
+    let littleEndianKey = new Uint32Array([
```

Comment:
Also, we use Int32Array elsewhere in this file because performance testing in v8 indicated that Uint32Array was causing bounds checking around accesses and that caused the hot inner loop to get de-optimized. Have you done any testing of this change against the old implementation? I know this level of performance scrutiny probably seems like overkill but this code has been identified as a bottleneck on playback framerate in the past.